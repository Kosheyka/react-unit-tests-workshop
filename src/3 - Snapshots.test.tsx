import React, { FC } from 'react';
import { create } from 'react-test-renderer';

const BaseButton: FC = () => {
    return <div className='button__wrapper'>
        <button>Click me</button>
    </div>;
};

export interface IMessage {
    id: string;
    entityType: unknown;
    text: string;
    isRead: boolean;
    createdAt?: {
        date: string;
        time: string;
    };
    attachments?: unknown;
}

interface IComponent {
    title: string,
    message: IMessage,
}

export const Component: FC<IComponent> = ({ title, message }) => {
    return <div>
        <h1>{title}</h1>
        <h3>{message.text}</h3>
        <BaseButton />
    </div>;
};

it('Should render correctly', () => {
    const messageStub: IMessage = {
        text: 'qwewqewq',
    } as IMessage;

    const component = create(<Component title={'hello'} message={messageStub} />);
    expect(component.toJSON()).toMatchSnapshot();
});
