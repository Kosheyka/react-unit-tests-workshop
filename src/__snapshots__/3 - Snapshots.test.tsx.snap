// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div>
  <h1>
    hello
  </h1>
  <h3>
    qwewqewq
  </h3>
  <div
    className="button__wrapper"
  >
    <button>
      Click me
    </button>
  </div>
</div>
`;
